File: jdi tests/jdi-uitest-webtests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/pages/ContactForm.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.pages;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.testing.unittests.entities.Contact;
import com.epam.jdi.uitests.web.selenium.elements.common.TextArea;
import com.epam.jdi.uitests.web.selenium.elements.common.TextField;
import com.epam.jdi.uitests.web.selenium.elements.composite.Form;
import org.openqa.selenium.support.FindBy;

import java.util.Arrays;
import java.util.List;

/**
 * Created by Dmitry_Lebedev1 on 10/16/2015.
 */
public class ContactForm extends Form<Contact> implements IContactForm {
    @FindBy(id = "Name")
    public TextField name;
    @FindBy(id = "LastName")
    public TextField lastName;
    @FindBy(id = "Description")
    public TextArea description;

    @FindBy(xpath = "//*[text()='Submit']")
    public IButton submit;

    public List<String> getFormValue() {
        return Arrays.asList(
                name.getText(),
                lastName.getText(),
                description.getText());
    }

    public void fillForm(Contact contact) {
        name.newInput(contact.name);
        lastName.newInput(contact.lastName);
        description.newInput(contact.description);
    }
}

File: Python/Test/jdi_uitests_webtests/main/page_objects/sections/contact_form.py
from JDI.web.selenium.elements.api_interact.find_element_by import By
from JDI.web.selenium.elements.common.button import Button
from JDI.web.selenium.elements.common.text_area import TextArea
from JDI.web.selenium.elements.common.text_field import TextField
from JDI.web.selenium.elements.composite.form import Form


class ContactForm(Form):
    def __init__(self, by_locator=None):
        super(ContactForm, self).__init__(by_locator)

    first_name = TextField(By.id("Name"))
    last_name = TextField(By.id("LastName"))
    description = TextArea(By.id("Description"))

    submit = Button(By.xpath("//*[text()='Submit']"))

    def get_form_value(self):
        return {self.first_name.get_text(), self.last_name.get_text(), self.description.get_text()}


class ContactFormTwoButtons(Form):
    def __init__(self, by_locator=None):
        super(ContactFormTwoButtons, self).__init__(by_locator)

    first_name = TextField(By.id("Name"))
    last_name = TextField(By.id("LastName"))
    description = TextArea(By.id("Description"))

    submit = Button(By.xpath("//*[text()='Submit']"))
    calculate = Button(By.xpath("//*[text()='Calculate']"))

    def get_form_value(self):
        return {self.first_name.get_text(), self.last_name.get_text(), self.description.get_text()}